File: product-services/src/main/java/com/microservices/productservices/service/ProductGallaryService.java
package com.microservices.productservices.service;

import com.microservices.productservices.payload.request.ProductGallaryRequest;
import com.microservices.productservices.payload.response.ProductGallaryResponse;

import java.util.List;
import java.util.UUID;

public interface ProductGallaryService {

    ProductGallaryResponse create(ProductGallaryRequest productGallaryRequest);

    void setImage(UUID id, String image);

    ProductGallaryResponse getById(UUID id);

    List<ProductGallaryResponse> getAll();

    ProductGallaryResponse update(UUID id, ProductGallaryRequest productGallaryRequest);

    ProductGallaryResponse delete(UUID id);

    void deleteByProductId(UUID productId);

    List<ProductGallaryResponse> findByProductId(UUID productId);
}

File: reactjs-app/src/services/ProductGallaryService.js
import httpAxios from '../httpAxios';

const ProductGallaryService = {
    create: (data) => {
        return httpAxios.post(`product-services/api/product-galleries/create`, data);
    },
    setImage: (data) => {
        return httpAxios.put(`product-services/api/product-galleries/set-image`, data);
    },
    getById: (id) => {
        return httpAxios.get(`product-services/api/product-galleries/get-by-id/${id}`);
    },
    getByProductId: (id) => {
        return httpAxios.get(`product-services/api/product-galleries/get-by-product-id/${id}`);
    },
    getAll: () => {
        return httpAxios.get(`product-services/api/product-galleries/get-all`);
    },
    delete: (id) => {
        return httpAxios.delete(`product-services/api/product-galleries/delete/${id}`);
    },
    deleteByProductId: (productId) => {
        return httpAxios.delete(`product-services/api/product-galleries/delete-by-product-id/${productId}`);
    },
}
export default ProductGallaryService;
